lib/puppet/parser/functions/clamp.rb

```diff
@@ -0,0 +1,31 @@
+#
+# clamp.rb
+#
+
+module Puppet::Parser::Functions
+  newfunction(:clamp, :type => :rvalue, :doc => <<-EOS
+    Clamps value to a range.
+    EOS
+  ) do |args|
+
+    raise(Puppet::ParseError, "clamp(): Wrong number of arguments " +
+          "need at least one") if args.size < 1
+
+    result = []
+
+    args.each do |x|
+      if x.is_a?(Array)
+        result += x
+      else
+        result << x
+      end
+    end
+
+    raise(Puppet::ParseError, "clamp(): Wrong number/type of arguments " +
+          "need three to clamp") if result.size != 3
+
+    result.map{ |n| n.to_i }.sort[1]
```

Comment:
Using of "sort" was a main goal of that function otherwise we could use mix/max now.
I believe sort() is so intuitive/native as far as it might be (get middle value by sort[1]).